src/Data/ProgrammesDb/EntityRepository/CollapsedBroadcastRepository.php

```diff
@@ -167,6 +167,30 @@ public function findByCategoryAncestryAndStartAtDateRange(
         return $this->resolveProgrammeParents($result);
     }
 
+    public function findByProgramme(
+        int $programmeId,
+        bool $isWebcastOnly,
+        ?int $limit,
+        int $offset
+    ): array {
+        $qb = $this->createQueryBuilder('collapsedBroadcast', false)
+            ->addSelect(['programmeItem', 'image', 'masterBrand', 'mbNetwork'])
+            ->leftJoin('programmeItem.image', 'image')
+            ->leftJoin('programmeItem.masterBrand', 'masterBrand')
+            ->leftJoin('masterBrand.network', 'mbNetwork')
+            ->andWhere('collapsedBroadcast.isWebcastOnly = :isWebcastOnly')
+            ->andWhere('programmeItem.id = :programmeItemId')
+            ->setParameter('programmeItemId', $programmeId)
+            ->setParameter('isWebcastOnly', $isWebcastOnly)
+            ->addOrderBy('collapsedBroadcast.endAt', 'ASC')
+            ->setFirstResult($offset)
+            ->setMaxResults($limit);
+
+        $result = $qb->getQuery()->getResult(Query::HYDRATE_ARRAY);
+
+        return $this->explodeFields($result);
```

Comment:
I think you should probably still be resolving programme parents here, unless you're very sure you don't need them

Reply:
I don't need them. I don't know if someone will in the future.

Reply:
If you don't resolve the parents then things like image and masterbrand inheritance on the ProgrammeItem will not work correctly (as it won't have a parent to check for an image's existence).

If you don't need the ProgrammeItem that is attached to the CollapsedBroadcast at all (which sounds like it might be the case as you're already filtering by it), then don't bother joining to programmeItem (and thus), Image, MasterBrand and Network.

Reply:
Looking at it again, it's probably also a good idea to have this sort by start date, just because there's already an index on that field, and it should make bugger all difference to the output.